### PR TITLE
Data scripts

### PR DESCRIPTION
**Pull request by iamsitting**
#983

### CODE REVIEW

File: ap/attendance/management/commands/data_check_rolls.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  # to use: python ap/manage.py data_check_rolls --settings=ap.settings.dev
+  def _mislink_rolls(self):
+    # Pulls all rolls that has a mislink, the event that the rolls points to does not exist in the trainee's active schedule
+    right_now = datetime.now().strftime("%m%d%Y_%H%M%S")
+    f = open('../mislink_rolls' + right_now + '.txt', 'w')
+    output = '{0}: {1}-- Submitted by: {2} \n'
+    rolls = Roll.objects.all().order_by('event__id', 'date')
+    for r in rolls:
+      try:
+        trainees = r.event.schedules.all().values('trainees')
```
Comment (yipingbenjaminliu) [inline ap/attendance/management/commands/data_check_rolls.py]: it's tricky here cause you need to make sure the event exists on the right weeks. there may be a roll with date in the 5th week but the schedule only has that event up until the 3rd week

File: ap/attendance/management/commands/data_check_rolls.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        slips = r.leaveslips.all()
+        if slips.count() == 0:
+          print output.format(r.id, r, r.submitted_by)
+          f.write(output.format(r.id, r, r.submitted_by))
+      except Exception as e:
+        print output.format(r.id, e, r.submitted_by)
+        f.write(output.format(r.id, e, r.submitted_by))
+    f.close()
+
+  def _mislink_leaveslips(self):
```
Comment (yipingbenjaminliu) [inline ap/attendance/management/commands/data_check_rolls.py]: also do a pull to see if there's a roll for trainee X that matches the attached roll and print out a line saying possible rolls

File: ap/attendance/management/commands/data_check_rolls.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        if r.trainee.id not in tr_ids:
+          print output.format(r.id, r, r.submitted_by)
+          f.write(output.format(r.id, r, r.submitted_by))
+
+      except Exception as e:
+        print output.format(r.id, e, r.submitted_by)
+        f.write(output.format(r.id, e, r.submitted_by))
+    f.close()
+
+  def _ghost_rolls(self):
```
Comment (yipingbenjaminliu) [inline ap/attendance/management/commands/data_check_rolls.py]: check to see if there's a leaveslip submitted by the trainee for other rolls or events on the date that this roll takes place

**Comment by iamsitting** [comment]
@yipingbenjaminliu Check it out now.

File: ap/attendance/management/commands/data_check_rolls.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+def out(s, f):
+  print s
+  f.write(s)
+
+
+class Command(BaseCommand):
+  # to use: python ap/manage.py data_check_rolls --settings=ap.settings.dev
+
+  def _mislink_rolls(self):
```
Comment (yipingbenjaminliu) [inline ap/attendance/management/commands/data_check_rolls.py]: make sure you're looking at the schedule with the highest priority. If the event exists on a schedule with a lower priority, this script won't pull that row.

File: ap/attendance/management/commands/data_check_rolls.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            if r.trainee.id not in s.trainees.values_list('id', flat=True):
+              out('Trainee DNM: ', f)
+              out(output.format(str(r.id), r, r.submitted_by), f)
+              break
+            else:
+              if str(ct.term_week_of_date(r.date)) not in s.weeks:
+                out('Wrong Week: ', f)
+                out(output.format(str(r.id), r, r.submitted_by), f)
+                break
+        except Exception as e:
```
Comment (yipingbenjaminliu) [inline ap/attendance/management/commands/data_check_rolls.py]: also make a function that pulls possible events that the roll should be attached to by looking at the attached event's start and end time or name or event type

**Pull request merged**